crates/oxc_linter/src/rules/vitest/prefer_spy_on.rs

```diff
@@ -0,0 +1,306 @@
+    /// `mockFn.mockRestore()` or by setting `restoreMocks` to `true` in the Vitest
+    /// config.
+    ///
+    /// Note: The mock created by `vi.spyOn()` still behaves the same as the original
+    /// function. The original function can be overwritten with
+    /// `mockFn.mockImplementation()` or by some of the
+    /// [other mock functions](https://vitest.dev/api/mock.html).
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Directly overwriting properties with mock functions can lead to cleanup issues
+    /// and test isolation problems. When you manually assign a mock to a property,
+    /// you're responsible for restoring the original implementation, which is easy to
+    /// forget and can cause tests to interfere with each other. Using `vi.spyOn()`
+    /// provides automatic cleanup capabilities and makes your tests more reliable.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```javascript
+    /// Date.now = vi.fn();
+    /// Date.now = vi.fn(() => 10);
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```javascript
+    /// vi.spyOn(Date, 'now');
+    /// vi.spyOn(Date, 'now').mockImplementation(() => 10);
+    /// ```
+    PreferSpyOn,
+    vitest,
+    style,
+    fix,
+);
+
+impl Rule for PreferSpyOn {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::AssignmentExpression(assign_expr) = node.kind() else {
+            return;
+        };
+
+        let left = &assign_expr.left;
+        let right = &assign_expr.right;
+
+        let Some(left_assign) = left
+            .as_simple_assignment_target()
+            .and_then(SimpleAssignmentTarget::as_member_expression)
+        else {
+            return;
+        };
+
+        match right {
+            Expression::CallExpression(call_expr) => {
+                Self::check_and_fix(assign_expr, call_expr, left_assign, node, ctx);
+            }
+            _ => {
+                if let Some(mem_expr) = right.as_member_expression() {
+                    let Expression::CallExpression(call_expr) = mem_expr.object() else {
+                        return;
+                    };
+                    Self::check_and_fix(assign_expr, call_expr, left_assign, node, ctx);
+                }
+            }
+        }
+    }
+}
+
+impl PreferSpyOn {
+    fn check_and_fix<'a>(
+        assign_expr: &AssignmentExpression,
+        call_expr: &'a CallExpression<'a>,
+        left_assign: &MemberExpression,
+        node: &AstNode<'a>,
+        ctx: &LintContext<'a>,
+    ) {
+        let Some(vi_fn_call) =
+            parse_general_jest_fn_call(call_expr, &PossibleJestNode { node, original: None }, ctx)
+        else {
+            return;
+        };
+        let Some(first_fn_member) = vi_fn_call.members.first() else {
+            return;
+        };
+
+        if first_fn_member.name().unwrap() != "fn" {
+            return;
+        }
```

Comment:
Don't call `unwrap`, as it can crash the linter in some cases. Add a `let-else` instead like this:

```suggestion
        let Some(first_fn_member_name) = vi_fn_call.members.first().map(|member| member.name()) else {
            return;
        };

        if first_fn_member_name != "fn" {
            return;
        }
```